src/main/java/io/github/deweyjose/graphqlcodegen/Validations.java

```diff
@@ -0,0 +1,61 @@
+package io.github.deweyjose.graphqlcodegen;
+
+import java.io.File;
+import java.util.*;
+
+public class Validations {
+
+  /**
+   * We sort the input files by their paths to ensure a consistent order for processing.
+   *
+   * We maintain two sets:
+   *   encounteredDirectories to keep track of encountered directories
+   *   encounteredPaths to keep track of encountered paths.
+   *
+   * We iterate through the sorted files and perform the following checks:
+   *   Directory overlap: throw an exception if duplicate directories are configured.
+   *   Duplicate files: throw an exception if duplicate files are configured.
+   *   Ancestry in already added directories:
+   *    if files exist in configured directories or subdirectories of configured directories
+   *    throw an exception
+   * @param files
+   */
+  public static void verifySchemaPaths(Collection<File> files) {
+    // Sort the input files to ensure consistent order
+    List<File> sortedFiles = new ArrayList<>(files);
+    sortedFiles.sort(Comparator.comparing(File::getPath));
+
+    Set<File> encounteredDirectories = new HashSet<>();
+    Set<String> encounteredPaths = new HashSet<>();
+
+    for (File file : sortedFiles) {
+      String path = file.getPath();
+
+      // Check for directory overlap
+      File parent = file.getParentFile();
+
+      // Check for ancestry in already added directories
+      while (parent != null) {
+        if (encounteredDirectories.contains(parent)) {
+          throw new IllegalArgumentException(
+            String.format("Overlap configured: %s contains %s", parent.getPath(), file.getPath())
+          );
+        }
+        parent = parent.getParentFile();
+      }
+
+      // Update the encountered sets
+      if (file.isDirectory()) {
+        encounteredDirectories.add(file);
+      }
+
+      // keep it simple, never allow duplicate configuration
+      if (encounteredPaths.contains(path)) {
+        throw new IllegalArgumentException(
+          String.format("Duplicate path configured: %s", path)
```

Comment:
@mjfaga refactored things a bit. This catches the case where folks misconfigured by adding `/dir` more than once or `/dir/a.graphqls` more than once.

My intention was just validate the config (sorry for the confusion, at one point I thought about being fancier and de-duping but wanted to simplify).

Reply:
I'm not opposed really to shifting back to an approach where we collect and return a valid set of paths if that exists.

Reply:
actually... this get reduced to Set<File> when it's passed to DGS anyway. I'll remove this, it's not really needed. It's odd to configure like that, but we don't need to throw here.